source/NVDAObjects/JAB/__init__.py

```diff
@@ -246,7 +246,12 @@ def _get_keyboardShortcut(self):
 				continue
 			keyList=[]
 			# We assume alt  if there are no modifiers at all and its not a menu item as this is clearly a nmonic
-			if (binding.modifiers&JABHandler.ACCESSIBLE_ALT_KEYSTROKE) or (not binding.modifiers and self.role!=controlTypes.ROLE_MENUITEM):
+			altModifier = binding.modifiers&JABHandler.ACCESSIBLE_ALT_KEYSTROKE
+			if not altModifier:
+				if not binding.modifiers and self.role!=controlTypes.ROLE_MENUITEM:
+					parent=self.parent
+					altModifier=isinstance(parent,JAB) and parent.role==controlTypes.ROLE_MENUBAR
+			if altModifier:

```

Comment:
Can this just be an else?

Reply:
NO, as in the if block above this one, the value of altModifier can be changed. We need to reevaluate it